### ISSUE TITLE
Add Support for Out-of-source Builds

### ISSUE THREAD

**Issue by Nopey** [SLikeSoft/SLikeNet#12]
Currently when SLikeNet is built it installs the headers & archives in the source directory.
This prevents people who are building it from specifying where to install it. (making packages is more difficult then it need be)
here is an example of code that is installing the static library inside the source directory
(From LibStatic/CMakeLists.txt)
```cmake
	target_link_libraries (RakNetLibStatic ${RAKNET_LIBRARY_LIBS})
	INSTALL(TARGETS RakNetLibStatic DESTINATION ${RakNet_SOURCE_DIR}/Lib/RakNetLibStatic)
	INSTALL(FILES ${ALL_HEADER_SRCS} DESTINATION ${RakNet_SOURCE_DIR}/include/raknet)
```

**Comment by Luke1410** [comment]
Thanks for the report. This is already on our list (internal number SLNET-54). I'll prioritize this now for the current development version since it makes sense to resolve this alongside SLNET-115 (i.e. see #11)

**Comment by Luke1410** [comment]
This has been done now and is available in the SVN and GitHub Repositories. Using CMake on non-Windows targets will install the libs to CMAKE_INSTALL_PREFIX/lib and headers to CMAKE_ISNTALL_PREFIX/include now.

Please note that this change is currently planned to be shipped with 0.2.0 atm (not with the upcoming 0.1.1 version, since it might break existing build environments relying on the files being installed in the source directories).
